### ISSUE TITLE
a='DEADBEEF'XL works in IDL but not in GDL (TEST_ROUTINE_NAMES fails on CI as a result)

### ISSUE THREAD

**Issue by slayoo** [gnudatalanguage/gdl#1188]
```
% Long integer constant must be less than 2147483648.
  At: /home/runner/work/gdl/gdl/testsuite/test_routine_names.pro
% Procedure not found: TEST_ROUTINE_NAMES
```
e.g. here: https://github.com/gnudatalanguage/gdl/runs/4147065434?check_suite_focus=true

**Comment by slayoo** [comment]
@acoulais @GillesDuvert 
I believe, this is a regression.
The minimal reproducer is:
```
GDL> a='DEADBEEF'XL                                          
% Long integer constant must be less than 2147483648.
```
Does it work with IDL?

Also, GDL fails to report which line the error comes from when executing the `test_routine_names` - does IDL tell line no. in such case?

**Comment by alaingdl** [comment]
it is OK in IDL 8.4 on U14.04

```
IDL> a='DEADBEEF'XL    
IDL> a
  -559038737
```

HTH

**Comment by slayoo** [comment]
thanks @alaingdl

**Comment by GillesDuvert** [comment]
a regression indeed.
bug at line 222 of dnode.cpp
This routine forgets about negative numbers ?

**Comment by GillesDuvert** [comment]
solved in #1204
